addons/account/models/partner.py

```diff
@@ -292,6 +292,11 @@ class AccountFiscalPositionAccount(models.Model):
 class ResPartner(models.Model):
     _name = 'res.partner'
     _inherit = 'res.partner'
+    company_registry_number = fields.Char(string="Company Registry Number", help="The registry number of your company")
+
+    _sql_constraints = [
+        ('uniq_company_registry', 'unique(company_registry_number)', 'This company registry is already used by another partner. Please make sure it is unique.'),
+    ]
```

Comment:
Be careful that here you make it unique for the DB itself.
I think the spec wants it to be unique per country?

```suggestion
    _sql_constraints = [
        ('uniq_company_registry', 'unique(company_registry_number, country_id)', 'This company registry is already used by another partner. Please make sure it is unique.'),
    ]
```

Eventually, it could be unique per country and company, to avoid having it filled in a partner linked to BeCompany1 and not being able to have the same value for another partner in BeCompany2. (Altough one could argue that both companies could simply share the same partner, so I'm not sure about this one)